design.md

```diff
@@ -49,19 +49,95 @@ modules], [Renovate] needs to be modified to support the new external dependency
   - Support different version resolution logic for libraries (i.e., is depended upon by other
     projects) vs executable (i.e., is not depdended upon by other projects) repositories .
 
-<!-- Future Sections
 
 ## Design
 
-The following sections describe 
-
 ### Versioning
 
-#### Parsing and Sorting
+Versioning for Bazel modules has two components: the `version` string and the `compatibility_level`
+integer. The `version` string has two roles. First, it identifies a unique release for a Bazel
+module. Second, it acts as a sort field aiding in the ordering of releases. The
+`compatibility_level` groups releases identifying those which are compatible with each other.
+Together, these values dictate how to identify upgrade candidates for a Bazel module.
+
+#### Version Formats
+
+Bazel modules support a [relaxed SemVer specification]. Specifically, [the version documentation]
+states:
+
+> The version format we support is `RELEASE[-PRERELEASE][+BUILD]`, where `RELEASE`, `PRERELEASE`,
+> and `BUILD` are each a sequence of "identifiers" (defined as a non-empty sequence of ASCII
+> alphanumerical characters and hyphens) separated by dots. The `RELEASE` part may not contain
+> hyphens.
+
+Examples of valid `version` values are listed below:
+
+```sh
+1.2.3                 # SemVer
+7.0.0-pre.20230330.3  # SemVer with prerelease
+1.0+build2            # SemVer with two parts and a build
+20230125.2            # Date-patch format
+```
+
+#### Version Sorting
+
+To identify whether a release is an upgrade candidate, we need know how to order the version values.
+The Renovate manager will implement [the same Bazel module version sort] as is implemented in Bazel.
```

Comment:
That will be a `bazel` [versioning type](https://docs.renovatebot.com/modules/versioning/#supported-versioning) which essentially makes the `compatibility_level` the renovate "compatibility" value (instead of the default which is the semver-major)?

Reply:
That is correct. I will update the language to be clearer.